dirty_cat/_fuzzy_join.py

```diff
@@ -148,36 +151,53 @@ def fuzzy_join(
             f"analyzer should be either 'char', 'word' or 'char_wb', got {analyzer!r}",
         )
 
+    if how not in ["left", "right"]:
+        raise ValueError(
+            f"how should be either 'left' or 'right', got {how!r}",
+        )
+
     for param in [on, left_on, right_on]:
         if param is not None and not isinstance(param, str):
-            raise ValueError(
+            raise KeyError(
                 "Parameter 'left_on', 'right_on' or 'on' has invalid type, expected"
                 " string"
             )
 
-    left_table_clean = left.reset_index(drop=True).copy()
-    right_table_clean = right.reset_index(drop=True).copy()
+    if how == "left":
+        left_table_clean = left.reset_index(drop=True).copy()
+        right_table_clean = right.reset_index(drop=True).copy()
+    else:
+        # We inverse the process so that the join is performed on the right column
+        left_table_clean = right.reset_index(drop=True).copy()
+        right_table_clean = left.reset_index(drop=True).copy()
```

Comment:
Did we test that when we pass dataframes with an index, the function does approximately the same things as pd.merge?